pulp_smash/tests/pulp3/file/api_v3/test_crud_content_unit.py

```diff
@@ -0,0 +1,86 @@
+# coding=utf-8
+"""Tests that perform actions over content unit."""
+import unittest
+
+from requests.exceptions import HTTPError
+
+from pulp_smash import api, config, selectors, utils
+from pulp_smash.constants import FILE_URL
+from pulp_smash.tests.pulp3.constants import ARTIFACTS_PATH, FILE_CONTENT_PATH
+from pulp_smash.tests.pulp3.file.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+from pulp_smash.tests.pulp3.utils import clean_artifacts, get_auth
+
+
+class ContentUnitTestCase(unittest.TestCase, utils.SmokeTest):
+    """CRUD content unit.
+
+    This test targets the following issues:
+
+    * `Pulp #2872 <https://pulp.plan.io/issues/2872>`_
+    * `Pulp Smash #870 <https://github.com/PulpQE/pulp-smash/issues/870>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variable."""
+        cls.cfg = config.get_config()
+        clean_artifacts(cls.cfg)
+        cls.content_unit = {}
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        files = {'file': utils.http_get(FILE_URL)}
+        cls.artifact = cls.client.post(ARTIFACTS_PATH, files=files)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean class-wide variable."""
+        cls.client.delete(cls.artifact['_href'])
+
+    def test_01_create_content_unit(self):
+        """Create content unit."""
+        attrs = _gen_content_unit_attrs(self.artifact)
+        type(self).content_unit = self.client.post(FILE_CONTENT_PATH, attrs)
```

Comment:
This test doesn't verify that the `attrs` that are sent were accepted by Pulp. Try something like this:

```python
    def test_01_create_content_unit(self):
        """Create content unit."""
        attrs = _gen_content_unit_attrs(self.artifact)
        self.content_unit.update(self.client.post(FILE_CONTENT_PATH, attrs))
        for key, val in attrs.items():
            with self.subTest(key=key):
                self.assertEqual(self.content_unit[key], val)
```

Note that I call `self.content_unit.update()` instead of overwriting `type(self).content_unit`. Calling `update()` is safer, as it prevents certain classes of mistakes. Consider the following code:

```python
type(self).contennt_unit = self.client.post(FILE_CONTENT_PATH, attrs)
self.contennt_unit.update(self.client.post(FILE_CONTENT_PATH, attrs))
```

The former line of code will create a brand new attribute called `contennt_unit`, whereas the latter will throw an invalid attribute error.

Reply:
done.